### PR TITLE
dnsdist: Log Extended DNS Errors (EDE) to protobuf

### PR DESCRIPTION
**Pull request by rgacogne**
### Short description

<strike>Needs more testing and regression tests.</strike>

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [x] tested this code
- [x] included documentation (including possible behaviour changes)
- [ ] documented the code
- [x] added or modified regression test(s)
- [x] added or modified unit test(s)

### CODE REVIEW

**Comment by rgacogne** [comment]
Otto, I'm adding you as a reviewer in particular because Neil and I realized it would likely make sense to have this in the recursor as well, if you agree.

File: pdns/dnsdistdist/docs/rules-actions.rst [hunk 1]
```diff
@@ -1438,6 +1441,7 @@ The following actions exist.
   * ``serverID=""``: str - Set the Server Identity field.
   * ``ipEncryptKey=""``: str - A key, that can be generated via the :func:`makeIPCipherKey` function, to encrypt the IP address of the requestor for anonymization purposes. The encryption is done using ipcrypt for IPv4 and a 128-bit AES ECB operation for IPv6.
   * ``exportTags=""``: str - The comma-separated list of keys of internal tags to export into the ``tags`` Protocol Buffer field, as "key:value" strings. Note that a tag with an empty value will be exported as "<key>", not "<key>:". An empty string means that no internal tag will be exported. The special value ``*`` means that all tags will be exported.
+  * ``exportExtendedErrorsToMeta=""``: str - Export Extended DNS Errors present in the DNS response, if any, into the specified ``tags`` Protocol Buffer field. The EDE info code will be exported as an integer value, and the EDE extra text, if present, as a string value.
```
Comment (omoerbeek) [inline pdns/dnsdistdist/docs/rules-actions.rst:1444]:
```suggestion
  * ``exportExtendedErrorsToMeta=""``: str - Export Extended DNS Errors present in the DNS response, if any, into the ``meta`` Protocol Buffer field using the specified ``key``. The EDE info code will be exported as an integer value, and the EDE extra text, if present, as a string value.
```

File: pdns/dnsdistdist/dnsdist-edns.cc [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+std::pair<std::optional<uint16_t>, std::optional<std::string>> getExtendedDNSError(const PacketBuffer& packet)
+{
+  uint16_t optStart = 0;
+  size_t optLen = 0;
+  bool last = false;
+
+  int res = locateEDNSOptRR(packet, &optStart, &optLen, &last);
+
+  if (res != 0) {
+    return std::make_pair(std::nullopt, std::nullopt);
```
Comment (omoerbeek) [inline pdns/dnsdistdist/dnsdist-edns.cc:37]:
```suggestion
    return {std::nullopt, std::nullopt};
```
A few more instances below

**Review by omoerbeek — APPROVED** [review]

**Comment by rgacogne** [comment]
Suggestions applied, thanks!

**Pull request merged**